Gems/PhysX/Code/Editor/ColliderComponentMode.cpp

```diff
@@ -237,25 +259,28 @@ namespace PhysX
             &AzToolsFramework::ViewportUi::ViewportUiRequestBus::Events::CreateCluster, AzToolsFramework::ViewportUi::Alignment::TopLeft);
 
         // create and register the buttons
-        m_dimensionsModeButtonId = RegisterClusterButton(m_modeSelectionClusterId, "Scale");
-        m_offsetModeButtonId = RegisterClusterButton(m_modeSelectionClusterId, "Move");
-        m_rotationModeButtonId = RegisterClusterButton(m_modeSelectionClusterId, "Rotate");
+        m_buttonIds.resize(static_cast<size_t>(SubMode::NumModes));
+        m_buttonIds[static_cast<size_t>(SubMode::Offset)] = RegisterClusterButton(m_modeSelectionClusterId, "Move");
+        m_buttonIds[static_cast<size_t>(SubMode::Rotation)] = RegisterClusterButton(m_modeSelectionClusterId, "Rotate");
+        m_buttonIds[static_cast<size_t>(SubMode::Dimensions)] = RegisterClusterButton(m_modeSelectionClusterId, "Scale");
```

Comment:
I'd go the extra mile of doing an array with {"Move", "Rotate", "Scale"} matching the enum to fill `m_buttonIds` with a for loop over the enum index from 0 till SubMode::NumModes. that way you can also add extra protections, like an static asset that the array and m_buttonIds is the same (this will protect the code if the enum changes and let you know it's missing to add elements to `m_buttonIds`)

Reply:
I think it might make sense in the future for there to be buttons which don't activate modes, they just do something. For example, we might add a "fit to mesh" button. That would mean there wouldn't be a 1 to 1 mapping between buttons and modes.